corehq/apps/userreports/reports/view.py

```diff
@@ -274,7 +274,11 @@ def post(self, request, *args, **kwargs):
             raise Http403()
 
     def has_permissions(self, domain, user):
-        return True
+        if domain is None:
+            return False
+        if not user.is_active:
+            return False
+        return user.can_view_report(domain, 'corehq.reports.DynamicReport{}'.format(self.report_config_id))
```

Comment:
not blocking but maybe a helper function with a good doc string might help with future maintenance of this?